src/states/adult.rs

```diff
@@ -453,11 +454,19 @@ impl Adult {
 
         if self.in_authority(&signed_msg.routing_message().dst) {
             self.check_signed_message_integrity(&signed_msg)?;
-            self.routing_msg_backlog.push(signed_msg.clone());
+
+            match &signed_msg.routing_message().content {
+                MessageContent::GenesisUpdate(info) => {
+                    return self.handle_genesis_update(info.clone())
+                }
+                _ => {
+                    self.routing_msg_backlog.push(signed_msg.clone());
```

Comment:
`self.routing_msg_backlog.push(signed_msg.clone())` will be called twice for this branch?
I think here we only need: `_ => (),` ?